sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -323,6 +354,65 @@ public Instant renewMessageLock(ServiceBusReceivedMessage message) {
         return null;
     }
 
+    /**
+     * Disposes of the consumer by closing the underlying connection to the service.
+     */
+    @Override
+    public void close() {
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
+
+        final ArrayList<String> keys = new ArrayList<>(openConsumers.keySet());
+        for (String key : keys) {
+            removeLink(key, SignalType.ON_COMPLETE);
+        }
+
+        connectionProcessor.dispose();
+    }
+
+    private Mono<Boolean> isLockTokenValid(UUID lockToken) {
```

Comment:
Does this need to return a `Boolean`? It doesn't appear to be used by the caller, we just default into a `then`.

Reply:
Oh yes. Thanks for pointing that out. If it is false, it should just return.